### PR TITLE
Update the ABI to version 0.2.0 so that it works with Envoy 1.17

### PR DESCRIPTION
**Pull request by nullpo-head**
## Summary

* Merge the latest upstream's changes, which include the ABI updates
* Replaces our own mechanism for creating child contexts with an upstream one that has been independently merged.
  * Links: [Yaro’s mechanism](https://github.com/yskopets/proxy-wasm-rust-sdk/commit/59ca074aa8da94df108689e04c68befd1b19d94b) and [The upstream's one](https://github.com/proxy-wasm/proxy-wasm-rust-sdk/pull/34) 
  * Actually, Yaro's one seems better-designed to me, and there is a PR to replace the current upstream's mechanism with Yaro&s. So, this change could be reverted if that PR is merged. proxy-wasm/proxy-wasm-rust-sdk#67
 * Advertise the ABI version as `0.2.0`. Legacy functions such as `get_configuration` and new ones like `continue_stream` are now [switched according to the advertised version. ](https://github.com/proxy-wasm/proxy-wasm-cpp-host/blob/master/src/wasm.cc#L133-L135)
 * Fix `cargo outdated` job of the GitHub action so that it checks root deps only.

This PR and the accompanying change of `envoy-wasm-rust-sdk` allow wasm plugins to run on both `wasm:1.15` and `standard:1.17` version of Envoy.

### CODE REVIEW

File: src/types.rs [hunk 1]
```diff
@@ -49,6 +49,13 @@ pub enum Status {
     InternalFailure = 10,
 }
 
+#[repr(u32)]
+#[derive(Debug)]
```
Comment (yskopets) [inline src/types.rs:53]: Let's consistently use
```rust
#[derive(Copy, Clone, Eq, PartialEq, Hash, Debug)]
```
See https://rust-lang.github.io/api-guidelines/interoperability.html#types-eagerly-implement-common-traits-c-common-traits

File: src/types.rs [hunk 2]
```diff
@@ -90,3 +97,11 @@ pub enum StreamType {
     Request = 0,
     Response = 1,
 }
+
+#[repr(u32)]
+#[derive(Debug)]
```
Comment (yskopets) [inline src/types.rs]: Let's consistently use
```rust
#[derive(Copy, Clone, Eq, PartialEq, Hash, Debug)]
```
See https://rust-lang.github.io/api-guidelines/interoperability.html#types-eagerly-implement-common-traits-c-common-traits

File: src/types.rs (hunk 1 above)
Comment (yskopets) [inline src/types.rs]: Let's consistently use
```rust
#[derive(Copy, Clone, Eq, PartialEq, Hash, Debug)]
```
See https://rust-lang.github.io/api-guidelines/interoperability.html#types-eagerly-implement-common-traits-c-common-traits

**Review by yskopets — APPROVED** [review]
There are 2 action points I'd like you to take.

Otherwise, awesome!

Thanks a lot!

**Comment by nullpo-head** [comment]
Please don't merge this yet for a while. I've messed up commits when I fixed my commits' email address and signature.

File: src/types.rs (hunk 2 above)
Comment (nullpo-head) [reply src/types.rs]: Fixed. Thanks!

File: src/types.rs (hunk 1 above)
Comment (nullpo-head) [reply src/types.rs]: Fixed. Thanks!

**Comment by nullpo-head** [comment]
> Please don't merge this yet for a while. I've messed up commits when I fixed my commits' email address and signature.

Now they're fixed.

**Review by yskopets — APPROVED** [review]

**Comment by yskopets** [comment]
@nullpo-head Thank you!

**Pull request merged**